### PR TITLE
[fix][doc] Move `Publish to partitioned topics` section to the client libraries

### LINKED ISSUE #17101: [Doc]  admin-api-topics contains wrong paragraph for concepts-messaging
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### What issue do you find in Pulsar docs?

The admin-api-topics page contains information on how to administrate topics using the admin API.
Towards the end there is a section that explains using the Client APIs Producer to write to a partitioned-topic.
https://pulsar.apache.org/docs/next/admin-api-topics/#publish-to-partitioned-topics
This section is in parts a duplicate of https://pulsar.apache.org/docs/next/concepts-messaging/#partitioned-topics but also contains some more information that does not belong on the Admin API page.

### What is your suggestion?

This section is in parts a duplicate of https://pulsar.apache.org/docs/next/concepts-messaging/#partitioned-topics but also contains some more information. I believe this section should be removed from the admin API page and maybe parts of it should be merged into the concepts-messaging page.

### Any reference?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by RobertIndie**
Fixes #17101

### Motivation

The `Publish to partitioned topics` section should not appear in the admin API page.

### Modifications

* Move `Publish to partitioned topics` section to the client libraries
* Remove `cookbooks-partitioned.md`
* Remove all links to `cookbooks-partitioned.md`

image

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -596,9 +596,98 @@ Producer<byte[]> producer = client.newProducer()
 
 ```
 
-### Message routing
+### Publish to partitioned topics
 
-When using partitioned topics, you can specify the routing mode whenever you publish messages using a producer. For more information on specifying a routing mode using the Java client, see the [Partitioned Topics cookbook](cookbooks-partitioned.md).
+By default, Pulsar topics are served by a single broker, which limits the maximum throughput of a topic. *Partitioned topics* can span multiple brokers and thus allow for higher throughput.
+
+You can publish to partitioned topics using Pulsar client libraries. When publishing to partitioned topics, you must specify a routing mode. If you do not specify any routing mode when you create a new producer, the round robin routing mode is used.
```
Comment (momo-jun) [inline site2/docs/client-libraries-java.md:603]:
```suggestion
You can publish messages to partitioned topics using Pulsar client libraries. When publishing messages to partitioned topics, you must specify a routing mode. If you do not specify any routing mode when you create a new producer, the round-robin routing mode is used.
```
I think an object is required here and in the heading to indicate where the action will behave.

File: site2/docs/client-libraries-java.md [hunk 2]
```diff
@@ -596,9 +596,98 @@ Producer<byte[]> producer = client.newProducer()
...
+
+#### Routing mode
+
+You can specify the routing mode in the ProducerConfiguration object that you use to configure your producer. The routing mode determines which partition(internal topic) that each message should be published to.
```
Comment (momo-jun) [inline site2/docs/client-libraries-java.md:607]:
```suggestion
You can specify the routing mode in the ProducerConfiguration object that you use to configure your producer. The routing mode determines which partition (internal topic) each message should be published to.
```

File: site2/docs/client-libraries-java.md [hunk 3]
```diff
@@ -596,9 +596,98 @@ Producer<byte[]> producer = client.newProducer()
...
+
+The following {@inject: javadoc:MessageRoutingMode:/client/org/apache/pulsar/client/api/MessageRoutingMode} options are available.
+
+Mode     | Description
+:--------|:------------
+`RoundRobinPartition` | If no key is provided, the producer publishes messages across all partitions in round-robin policy to achieve the maximum throughput. Round-robin is not done per individual message, round-robin is set to the same boundary of batching delay to ensure that batching is effective. If a key is specified on the message, the partitioned producer hashes the key and assigns message to a particular partition. This is the default mode.
```
Comment (momo-jun) [inline site2/docs/client-libraries-java.md:613]:
```suggestion
`RoundRobinPartition` | If no key is provided, the producer publishes messages across all partitions in the round-robin policy to achieve the maximum throughput. Round-robin is not done per individual message, round-robin is set to the same boundary of batching delay to ensure that batching is effective. If a key is specified on the message, the partitioned producer hashes the key and assigns the message to a particular partition. This is the default mode.
```

**Review by momo-jun — APPROVED** [review]
Nice fix.

**Comment by momo-jun** [comment]
@RobertIndie Maybe it can catch up with the 2.11 milestone if the doc set has not been baselined.

File: site2/docs/client-libraries-java.md (hunk 3 above)
Comment (Anonymitaet) [inline site2/docs/client-libraries-java.md:613]:
```suggestion
`RoundRobinPartition` | If no key is provided, the producer publishes messages across all partitions in the round-robin policy to achieve the maximum throughput. Round-robin is not done per individual message. It is set to the same boundary of batching delay to ensure that batching is effective. If a key is specified on the message, the partitioned producer hashes the key and assigns the message to a particular partition. This is the default mode.
```

File: site2/docs/client-libraries-java.md [hunk 4]
```diff
@@ -596,9 +596,98 @@ Producer<byte[]> producer = client.newProducer()
...
+
+public class AlwaysTenRouter implements MessageRouter {
+    public int choosePartition(Message msg) {
+        return 10;
+    }
+}
+
+```
+
+With that implementation, you can send
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-java.md:657]:
```suggestion
With that implementation, you can send messages to partitioned topics as below.
```

send what? Is this correct?

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**